Framework/Core/include/Framework/SerializationMethods.h

```diff
@@ -0,0 +1,66 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef FRAMEWORK_SERIALIZATIONMETHODS_H
+#define FRAMEWORK_SERIALIZATIONMETHODS_H
+
+/// @file SerializationMethods.h
+/// @brief Type wrappers for enfording a specific serialization method
+
+namespace o2
+{
+namespace framework
+{
+
+/// @class SerializationMethod
+/// A base class for the serialization method type converters
+/// Type converters wrap a type for which a method of serialization is
+/// enforced in a function call.
+/// Currently the main purpose of the base class is to make the converters
+/// polymorphic and distinguishable from the messageable types in
+/// automatic substitutions
+class SerializationMethod
```

Comment:
I would make it a struct and avoid any virtual method. Why do you need them? Actually, why do you need this at all? Given they are all templated, the various serialization methods do not need any polymorphism.

Reply:
In order to avoid ambiguous substitutions I had to make this converter non- messageable. The  comment just in front of the definition tries to explain this.

We would need to add is-not-ROOTSerialized like statements in all the specializations and possibly the same for other converters coming in the future. The simple solution is to make the converters non-messageable, but maybe there are better ways.

Reply:
Ah, ok, I see. How about we check for a `::non_messageable` type when checking if `is_messageable`. Kinda like when you do `is_container` by checking the types which are known to be required for a `std::container`.

Reply:
Actual example of what I mean:

```C++
#include <iostream>
#include <type_traits>

struct NonMessageable {};

struct is_non_messageable {
  template <typename T>
  static typename std::enable_if<std::is_same<typename T::non_messageable, NonMessageable>::value, bool>::type docheck(int) {
    return true;
  }
  template <typename T>
  static bool docheck(long) {
    return false;
  }
};

struct Foo {
  using non_messageable = NonMessageable;
};

int main(int argc, char**argv) {
  std::cerr << is_non_messageable::docheck<Foo>(0) << std::endl;
  std::cerr << is_non_messageable::docheck<int>(0) << std::endl;
}
```

can probably probably be included, but should give an idea of what I meant.